hkmc2/shared/src/main/scala/hkmc2/codegen/BlockTransformer.scala

```diff
@@ -238,3 +244,28 @@ class BlockTransformerShallow(subst: SymbolSubst) extends BlockTransformer(subst
         then b else HandleBlock(l2, res2, par2, args2, cls2, hdr2, bod, rst2)
     case _ => super.applyBlock(b)
 
+// does not traverse into any other block
```

Comment:
The semantics of this transformer seems broken wrt `Begin`. If you have `Begin{foo()}; bar()` you're not going to change `foo` even though you would have for the equivalent `foo(); bar()`.

Reply:
The job of this transformer is just to modify the paths and results in a block which do not require traversing another block to reach -- it's the job of the person using this transformer to traverse the sub-blocks, probably used in combination with one of the other block transformers.

As for `Begin`, both of its sub-blocks are expected to be traversed by whoever is using this transformer, not the transformer itself.

Reply:
Ok, then please pick a better name for this class and document in the code what you've just written here. The current comment is inadequate as documentation.

Reply:
Also it feels like a better architecture might be to use  a method called `applySubBlock` whenever a sub-block (not a top-level block) is traversed, which would mean you only have to override that here to return input unchanged, instead of all this boilerplate.

Reply:
Good point, I think that will be a better solution.